src/main/java/bio/terra/service/snapshot/SnapshotService.java

```diff
@@ -167,6 +167,29 @@ public SnapshotService(
     this.snapshotBuilderSettingsDao = snapshotBuilderSettingsDao;
   }
 
+  public String generateUpdatedSnapshotNameIfByRequest(SnapshotRequestModel model) {
+    SnapshotRequestContentsModel contentsModel = model.getContents().get(0);
```

Comment:
Could  `model.getContents()` throw an NPE?

Reply:
I think it could - but we already do this in lots of places. I think there's an argument that we should maybe add a helper method for `getFirstSourceOrThrow` that threw something better - depending on consensus, I could add that.

Reply:
maybe we could make a tech debt ticket for that work?

Reply:
Although, `get(0)` could throw `IndexOutOfBoundsException`. Adding `minItems: 1` to the openapi spec would fix that problem. (If you do this, I'd also add `maxItems: 1` since only one snapshot source is supported).

Reply:
https://broadworkbench.atlassian.net/browse/DC-1162